pkg/azuredisk/controllerserver_v2.go

```diff
@@ -140,10 +170,8 @@ func (d *DriverV2) CreateVolume(ctx context.Context, req *csi.CreateVolumeReques
 		}
 	}
 
-	if IsAzureStackCloud(d.cloud.Config.Cloud, d.cloud.Config.DisableAzureStackCloud) {
-		if maxShares > 1 {
-			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Invalid maxShares value: %d as Azure Stack does not support shared disk.", maxShares))
-		}
+	err := d.controllerProvisioner.ValidateControllerCreateVolumeParameters(parameters); err {
```

Comment:
Rather than expose the validation externally, pass the parameters to the provisioner and handle it internally. The parameters above a specific to Azure and may not apply to other supported platforms. Same with many of those below. I would expect that much of the logic in this method move to the provision, and simpluy unwraps the gRPC message to pass parameter to it.

Reply:
Modified cloud provisioner to expose only relevant functions.